### PR TITLE
[Aptos Data Client] Improve metrics and logic around polling.

### PR DESCRIPTION
**Pull request by JoshLind**
## Motivation

This PR updates the Aptos Data Client with several small tweaks:
- Add network and peer information for metrics when polling. Also, improve some of the logging and error messages.
- All storage requests should be sent to priority peers (if possible). This helps to reduce latency and increase reliability. Without this, we're seeing latency spikes/lag in the PFNs on devnet where the PFNs are requesting data from low quality peers that continue to timeout despite advertising the data (e.g., due to upload bandwidth constraints).
- Update the definition for priority peers to be based on node role, network id and connection origin. This is a better workaround given we're still working on topology awareness.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Unit tests and manual verification.

## Related PRs

None.

### CODE REVIEW

File: state-sync/aptos-data-client/src/aptosnet/mod.rs [hunk 1]
```diff
@@ -140,22 +144,15 @@ impl AptosNetDataClient {
...
-                if !priority_serviceable.is_empty() {
-                    priority_serviceable
-                } else {
-                    self.identify_serviceable(regular_peers, request)
-                }
-            } else {
-                let all_connected_peers = self.get_all_connected_peers()?;
-                self.identify_serviceable(all_connected_peers, request)
-            };
+        // All requests should be sent to prioritized peers (if possible).
```
[the remark this answers is not in the document]
Comment (JoshLind) [reply state-sync/aptos-data-client/src/aptosnet/mod.rs:147]: Aah, I noted this in the PR description:

> All storage requests should be sent to priority peers (if possible). This helps to reduce latency and increase reliability. Without this, we're seeing latency spikes/lag in the PFNs on devnet where the PFNs are requesting data from low quality peers that continue to timeout despite advertising the data (e.g., due to upload bandwidth constraints).

While it sounds nice from a load balancing point of view, what we're seeing today is that most of the public nodes are low quality and unable to actually serve data (within a reasonable enough timeout). At least now, we know that if nodes specify seed peers, they'll be prioritized 😄

**Review by sitalkedia — APPROVED** [review]
LGTM

File: state-sync/aptos-data-client/src/aptosnet/state.rs [hunk 2]
```diff
@@ -199,52 +203,72 @@ impl PeerStates {
...
     /// Marks the pending in-flight request as complete for the specified peer
     pub fn mark_in_flight_request_complete(&mut self, peer: &PeerNetworkId) {
-        if self.is_priority_peer(peer) {
-            let _ = self.in_flight_priority_polls.remove(peer);
-        } else {
-            let _ = self.in_flight_regular_polls.remove(peer);
-        };
+        // The priority of the peer might have changed since we
+        // last polled it, so we attempt to remove it from both
+        // the regular and priority in-flight requests.
```
Comment (msmouse) [inline state-sync/aptos-data-client/src/aptosnet/state.rs:237]: 👍

**Pull request closed without merging**